docs/source/_templates/layout.html

```diff
@@ -26,3 +26,9 @@
 {# Silence the navbar #}
 {% block docs_navbar %}
 {% endblock %}
+
+{# Add version warnings #}
+{% block footer %}
+  {{ super() }}
+  <script type="text/javascript" src="/docs/versionwarning.js"></script>
```

Comment:
There is a separate PR that adds the `versionwarnings.js` file: https://github.com/apache/arrow-site/pull/275 .
It has to be merged first.

Reply:
Why can't the file be added in `docs/source/_static` instead?

Reply:
And the link to the file: https://github.com/AlenkaF/arrow-site/blob/73fb93f1f2254efd11b83e8de8b020d085d009e7/docs/versionwarning.js

Reply:
> Why can't the file be added in `docs/source/_static` instead?

That is something Joris also suggested, but at the end decided it could stay as is. I do not mind changing, if it is preferred. cc @jorisvandenbossche

Reply:
I like `docs/source/_static/` because https://github.com/apache/arrow/blob/master/docs/source/_static/versions.json uses `docs/source/_static/`.

Reply:
Will move the JS file to `docs/source/_static/` as there seems to be a consensus on this topic.

Reply:
The question we have to answer here is whether we want to maintain this `versionwarning.js` file in the arrow repo or in the arrow-site repo. 
Because if we put it in `_static`, that folder is created by the output of sphinx, and so then we should include `versionwarning.js` in the sphinx source here in the arrow repo. On the other hand, if the goal is that it is a single file that supports all the different versions we currently include in the /docs in arrow-site, we can also maintain it in arrow-site (and if we maintain it there, I think it is easier to to put it in /docs instead of /docs/_static). This is the approach that mne-python and scikit-learn (on which the current versionwarning.js file is based) are using.

Reply:
In my opinion, I don't want to maintain files in `apache/arrow-site`'s `asf-site` branch as much as possible.
The branch is sometimes updated automatically. It may delete some files unexpectedly. See also: https://github.com/apache/arrow-site/pull/151
The branch has many large commits. It's difficult to follow history of maintained files.

Reply:
Ah, yes, that's a good point. I agree that we should avoid having the source only in the asf-site branch. I suppose that in theory we could still have the file in the arrow-site repo in the master branch and ensure this gets put in the docs/ repo on deployment. But that's also some additional complexity, so I suppose then we can just as well leave it in the arrow repo (where at least it lives in the docs/ repo).

We will just have to remember that whenever we want to do an update to the versionwarning.js file, we also have to port that patch to arrow-site to actually start using it (otherwise it only gets used once there is a new release)

Reply:
> We will just have to remember that whenever we want to do an update to the versionwarning.js file, we also have to port that patch to arrow-site to actually start using it (otherwise it only gets used once there is a new release)

I can add a warning comment in the `versionwarning.js` if that would help?

Reply:
Yes, that sounds good

Reply:
How about referring `/docs/dev/_static/versionwarning.js` instead of `/docs/_static/versionwarning.js`?

Reply:
BTW, I like `version-warnings.js` or `version_warnings.js` (I want word separator for readability)...

Reply:
> How about referring `/docs/dev/_static/versionwarning.js` instead of `/docs/_static/versionwarning.js`?

Ah, sorry, I just merged the PR on the site repo before checking the latest comments here. Now, if we decide to go with /dev/, then it should be easy to still update that in the arrow-site repo. 

While doing it in /dev/ would avoid two PRs, in practice you can't test a change in this repo though, and you always have to test it in arrow-site as well to check the changes. So then actually doing a PR there as well isn't too bad.

Reply:
I can change the name of the file and move it to `/docs/dev/_static/versionwarning.js`. As Joris mentioned, changes in the arrow-site repo will be needed afterwards.

Will wait a day or two to see if there is any other opinion/change of the name or file location and make the changes then.

Reply:
Can we use Sphinx variable for it something like the following?

```diff
diff --git a/ci/scripts/python_build.sh b/ci/scripts/python_build.sh
index cfac68bd6e..d02d654205 100755
--- a/ci/scripts/python_build.sh
+++ b/ci/scripts/python_build.sh
@@ -83,5 +83,10 @@ popd
 
 if [ "${BUILD_DOCS_PYTHON}" == "ON" ]; then
   ncpus=$(python -c "import os; print(os.cpu_count())")
-  sphinx-build -b html -j ${ncpus} ${arrow_dir}/docs/source ${build_dir}/docs
+  sphinx-build \
+    -b html \
+    -j ${ncpus} \
+    -D version_warning_js=://arrow.apache.org/docs/dev/_static/version_warning.js \
+    ${arrow_dir}/docs/source \
+    ${build_dir}/docs
 fi
diff --git a/docs/source/_templates/layout.html b/docs/source/_templates/layout.html
index 50511b1416..fd5186fdac 100644
--- a/docs/source/_templates/layout.html
+++ b/docs/source/_templates/layout.html
@@ -30,5 +30,5 @@
 {# Add version warnings #}
 {% block footer %}
   {{ super() }}
-  <script type="text/javascript" src="/docs/versionwarning.js"></script>
+  <script type="text/javascript" src="{{ pathto(version_warning_js, 1) }}"></script>
 {% endblock %}
diff --git a/docs/source/conf.py b/docs/source/conf.py
index 2c3686df87..92b7f2cbf4 100644
--- a/docs/source/conf.py
+++ b/docs/source/conf.py
@@ -148,6 +148,9 @@ autosummary_generate = True
 # The master toctree document.
 master_doc = 'index'
 
+# Path to version_warning.js.
+version_warning_js = "_static/version_warning.js"
+
 # General information about the project.
 project = u'Apache Arrow'
 copyright = f'2016-{datetime.datetime.now().year} Apache Software Foundation'
```

Reply:
> Can we use Sphinx variable for it something like the following?

Not sure, can try 👍

Reply:
@kou what would be the goal of that variable? So that you can build the docs locally with a different path (one that would match the local path)? 
That still won't help that you can test this locally when building the arrow docs, though, as it the javascript code needs the structure of different versions in subdirectories as we have in that arrow-site branch.

Reply:
Yes. I wanted to solve "in practice you can't test a change in this repo though, and you always have to test it in arrow-site as well to check the changes".

I thought that we can test almost features (I understand that we can't test all features) with this approach. We don't need to change `src=.../versionwaning.js` in all `.html` manually to test locally with this approach.

If this approach doesn't help `versionwarning.js` developers/maintainers, I'm OK without this approach. (I'm OK that they always need apache/arrow-site.)

Reply:
I propose to keep the current path to the JS file (docs/_static) and also the current way of injecting the src link in the html files. This way there will be no further changes needed in the arrow-site repo.

If we decide to move the file to the dev/ folder or use a variable in html (in my short attempt I wasn't able to make Sphinx work with variables) I think it is better to create a follow-up issue.

Reply:
Sounds good to me!

Reply:
I'm OK with it too.